### PR TITLE
mgmt/MCUmgr/img: Fix img_mgmt_get_unused_slot_area_id checks

### PR DESCRIPTION
**Pull request by de-nordic**
Remove image == -1 check that can not happen anymore.

### CODE REVIEW

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c [hunk 1]
```diff
@@ -240,7 +240,7 @@ static int img_mgmt_get_unused_slot_area_id(int image)
 {
 	int area_id = -1;
 
-	if (image == 0 || image == -1) {
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c:243]: `int image` should be uint

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c (hunk 1 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c:243]: Changed

**Review by nordicjm — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by carlescufi** [comment]
@de-nordic please rebase

**Comment by de-nordic** [comment]
@carlescufi Rebased.

**Review by nordicjm — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by carlescufi** [comment]
@de-nordic please rebase

**Pull request closed without merging**